staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go

```diff
@@ -224,35 +224,41 @@ func (p *Patcher) buildStrategicMergeFromBuiltins(gvk schema.GroupVersionKind, o
 
 // Patch tries to patch an OpenAPI resource. On success, returns the merge patch as well
 // the final patched object. On failure, returns an error.
-func (p *Patcher) Patch(current runtime.Object, modified []byte, source, namespace, name string, errOut io.Writer) ([]byte, runtime.Object, error) {
-	var getErr error
-	patchBytes, patchObject, err := p.patchSimple(current, modified, namespace, name, errOut)
+func (p *Patcher) Patch(info *resource.Info, modified []byte, errOut io.Writer) (bool, error) {
+	changed, patchObject, err := p.patchSimple(info.Object, modified, info.Namespace, info.Name, errOut)
 	if p.Retries == 0 {
 		p.Retries = maxPatchRetry
 	}
 	for i := 1; i <= p.Retries && apierrors.IsConflict(err); i++ {
 		if i > triesBeforeBackOff {
 			p.BackOff.Sleep(backOffPeriod)
 		}
-		current, getErr = p.Helper.Get(namespace, name)
-		if getErr != nil {
-			return nil, nil, getErr
+		info.Object, err = p.Helper.Get(info.Namespace, info.Name)
+		if err != nil {
+			return changed, err
 		}
-		patchBytes, patchObject, err = p.patchSimple(current, modified, namespace, name, errOut)
+		changed, patchObject, err = p.patchSimple(info.Object, modified, info.Namespace, info.Name, errOut)
 	}
 	if err != nil {
 		if (apierrors.IsConflict(err) || apierrors.IsInvalid(err)) && p.Force {
-			patchBytes, patchObject, err = p.deleteAndCreate(current, modified, namespace, name)
+			patchObject, err = p.deleteAndCreate(info.Object, modified, info.Namespace, info.Name)
 		} else {
-			err = cmdutil.AddSourceToErr("patching", source, err)
+			err = cmdutil.AddSourceToErr("patching", info.Source, err)
+		}
+
+		if err != nil {
+			return changed, err
 		}
 	}
-	return patchBytes, patchObject, err
+
+	info.Refresh(patchObject, true)
```

Comment:
Personally I don't think moving this here really contributes to code cleanliness. In fact I think it obscures the code a bit since now the `Patch` function now longer has somewhat functional semantics. Now it has side effects & modifies its arguments. 

And I think only one code path actually used a resource.Info and needed this.